lib/Sema/ConstraintSystem.cpp

```diff
@@ -622,10 +622,10 @@ ConstraintLocator *ConstraintSystem::getCalleeLocator(
   if (isExpr<ObjectLiteralExpr>(anchor))
     return getConstraintLocator(anchor, ConstraintLocator::ConstructorMember);
 
-  if (locator->isLastElement<LocatorPathElt::FunctionArgument>()) {
-    if (auto *CE = getAsExpr<CoerceExpr>(anchor)) {
-      return getConstraintLocator(CE->getSubExpr());
-    }
+  if (locator->findFirst<LocatorPathElt::CoercionOperand>()) {
+    locator = getConstraintLocator(simplifyLocatorToAnchor(locator));
```

Comment:
This doesn't seem correct... I think should simplify only anchor + CoercionOperand into the underlying operand expression and let the recursive call to `getCalleeLocator` handle the rest of the chain (if there is anything).

Reply:
Fixed =]

Reply:
This seems to cause regressions in two cases, it seems like not all subexpr should be returned as calleeLocator.

Reply:
I think it be simplified to the argument expression only if there are more elements after CoercionOperand?…

Reply:
The regression was in a very specific case where a requirement failure specific logic relies on callee locator to get information about context. Fixed in the last commit